packages/tools/client-debugger/client-debugger/src/FluidClientDebugger.ts

```diff
@@ -79,6 +101,28 @@ export class FluidClientDebugger
 			newState: ConnectionState.Disconnected,
 			timestamp: Date.now(),
 			clientId: undefined,
+			containerId: this.containerId,
+			changeKind: "disconnected",
+		});
+	};
+
+	private readonly containerClosedHandler = (): void => {
+		this._connectionStateLog.push({
+			newState: ConnectionState.Disconnected,
+			timestamp: Date.now(),
+			clientId: undefined,
+			containerId: this.containerId,
+			changeKind: "closed",
+		});
+	};
+
+	private readonly containerDisposeddHandler = (): void => {
+		this._connectionStateLog.push({
+			newState: ConnectionState.Connected,
```

Comment:
I think we probably need to replace the "connectionStateLog" with a more general "containerStateLog", since we're tracking more than just connection state changes here. Then we can push our "changeKind" values as `newState` instead of using `ConnectionState`.